pkg/channel/event_dispatcher.go

```diff
@@ -32,7 +32,18 @@ import (
 	"knative.dev/eventing/pkg/utils"
 )
 
-const correlationIDHeaderName = "Knative-Correlation-Id"
+const (
+	correlationIDHeaderName = "Knative-Correlation-Id"
+
+	// TODO make these constants configurable (either as env variables, config map, or part of broker spec).
+	//  Issue: https://github.com/knative/eventing/issues/1777
```

Comment:
I think this will become more and more relevant once you keep on tweaking things for performance. It will be easier if we add a configMap with defaults rather soonish... 
just a suggestion, but I don't mind if it's not done in this PR...

Reply:
I also add to @nachocano comment that a config map reader that checks these values and builds the http client could useful directly in https://github.com/knative/eventing/blob/master/pkg/kncloudevents/good_client.go , so eventing-contrib can use it too for sources...

Reply:
I initially thought about adding them directly in https://github.com/knative/eventing/blob/master/pkg/kncloudevents/good_client.go, but those functions are being called in many places, and most of them does not need this change.

For our current situation, I think we need this change in the fanout logic for Broker and Channel, a couple of Sources, (and probably the other Channel implementations?). And maybe we should support a global default configmap, as well as configmaps for each implementation. But I would rather leave them in the following PRs..

Reply:
Maybe to avoid any breakage, just add a new function like `NewClientFromConfigMap` (or something similar) and then we check for every use case if we want to create a client from config map or not